share/github-backup-utils/ghe-prune-snapshots

```diff
@@ -6,13 +6,26 @@ set -e
 # Bring in the backup configuration
 . $( dirname "${BASH_SOURCE[0]}" )/ghe-backup-config
 
+# Once we start pruning, this backup will no longer be valid.
+# So create or preserve its `incomplete` file and remove the
+# `incomplete` file last.
+prune_snapshot() {
+  local prune_dir
+  while read prune_dir; do
+    [ -n $prune_dir ] || return
+    touch "$prune_dir/incomplete"
+    find "$prune_dir" -not -path "$prune_dir/incomplete" | xargs rm -rf
```

Comment:
Oh bugger, I lost the rest of my suggestion 😢 . Lets try again.

Now for the completely outrageous. Rumour has it `rsync` can be quicker than `rm` to delete files, so it may be an idea to try something like this (it'll need testing as this is off the top of my head):

```
touch "$prune_dir/incomplete"
tempdir=$(mktemp -d -t backup-utils-prune-XXXXXX)
rsync -r --exclude "$prune_dir/incomplete" --delete "$tempdir/" "$prune_dir/"
rm -rf "$prune_dir" "$tempdir"
```

My biggest uncertainty with this approach is if it really is any quicker, and if the `--exclude` will stop the incomplete file from being deleted. It probably won't so we may need to drop an incomplete file into the `$tempdir` first.

Reply:
😱 Thank you for calling out min/max depth @lildude ! And for suggesting the rsync alternative.

I'll find or create a separate issue about `rsync` vs. `rm` for faster pruning, and I'll make the other edits you suggested here.